lib/auth.coffee

```diff
@@ -280,9 +295,10 @@ getAuth = (deps, opts) ->
 	# });
 	###
 	exports.getEmail = (callback) ->
-		token.getEmail().then (email) ->
-			throw new errors.ResinNotLoggedIn() if not email?
-			return email
+		getUserDetails()
+		.get ('email')
+		.catch ->
+			throw new errors.ResinNotLoggedIn()
```

Comment:
Why is this catch needed here? It doesn't match getUserId or whoami, and `getUserDetails` already throws this error.